ceremonies/src/lib.rs

```diff
@@ -1446,9 +1446,11 @@ impl<T: Config> Pallet<T> {
 	fn purge_community(cid: CommunityIdentifier) {
 		let current = <encointer_scheduler::Pallet<T>>::current_ceremony_index();
 		let reputation_lifetime = Self::reputation_lifetime();
-		for cindex in max(current - reputation_lifetime, 0)..current {
+
+		for cindex in max(current.saturating_sub(reputation_lifetime), 0)..current {
 			if cindex > reputation_lifetime {
-				Self::purge_registry(cindex - reputation_lifetime - 1);
+				let purge_index = current.saturating_sub(reputation_lifetime).saturating_sub(1);
```

Comment:
should be `cindex` instead of `current`

Reply:
Thanks, this review made me actually realize that the function before was not even correct. :)